### DISCUSSION TITLE
Bit Info Map from Atom Pair Fingerprint Generator isn't matching "on" FP bits

### DISCUSSION THREAD

**Discussion by luchini18** [rdkit/rdkit#5856] [Q&A] [answered]
Hi, I've been playing around with the rdFingerprintGenerators and found something that seems a bit off to me.
This is using rdkit version `2022.09.1`.

I'm concerned with stereochemistry and am interested in FP bits at these atom locations. 
I can generate Morgan fingerprints and when I compare the "on" bits to the keys in the BitInfoMap they match up, as expected. 
```
from rdkit import Chem
from rdkit.Chem import rdFingerprintGenerator

smiles = 'C[C@H]1CCC[C@@H](C)N1S(=O)(=O)c1ccc(N)cc1'
mol = Chem.MolFromSmiles(smiles)
stereo_atoms = [1,5]
nBits=2048

M_generator = rdFingerprintGenerator.GetMorganGenerator(radius=4,
                                                        includeChirality=True,
                                                        fpSize=nBits)
output = rdFingerprintGenerator.AdditionalOutput()
output.AllocateBitInfoMap()
fp = M_generator.GetFingerprint(mol,fromAtoms=stereo_atoms,additionalOutput=output)
bi = output.GetBitInfoMap()

print(list(bi.keys()))
print(list(fp.GetOnBits()))
```
output:
```
[769, 846, 972, 1019, 1836, 1837, 1880]
[769, 846, 972, 1019, 1836, 1837, 1880]
```
This is great for interpreting these bits, I can visualize them later with the built-in functions, and there is plenty of information in the docs for interpreting these bitinfo maps.
However, when I try the same routine for Atom Pair fingerprints these bit info keys do not match the on bits:
```
smiles = 'C[C@H]1CCC[C@@H](C)N1S(=O)(=O)c1ccc(N)cc1'
mol = Chem.MolFromSmiles(smiles)
stereo_atoms = [1,5]
nBits=2048

AP_generator = rdFingerprintGenerator.GetAtomPairGenerator(minDistance=1,
                                                           includeChirality=True,
                                                           fpSize=nBits)

output = rdFingerprintGenerator.AdditionalOutput()
output.AllocateBitInfoMap()
fp = AP_generator.GetFingerprint(mol,fromAtoms=stereo_atoms,additionalOutput=output)
bi = output.GetBitInfoMap()

print(list(bi.keys()))
print(list(fp.GetOnBits()))
```
output:
```
[8, 25, 116, 129, 183, 207, 222, 243, 268, 333, 348, 463, 468, 489]
[32, 33, 100, 464, 465, 466, 516, 517, 732, 733, 734, 828, 829, 888, 889, 972, 973, 974, 1072, 1073, 1332, 1333, 1392, 1393, 1852, 1853, 1872, 1873, 1956, 1957]
```

Is this expected? Does it make sense to look at bit info maps for atom pair fingerprints?
The format for these bit info maps also looks a bit different as well, I don't think it will output the atom and the radius, it looks like it is just pairs of atom indices. Was just wondering if there were resources for the differences in these maps and what information the bit info map for the atom pair fingerprints is telling me. Or if this is a bug I can submit an Issue... Does anybody have experience with these?

Thanks!

**Comment by ptosco** [comment, marked as answer]
@luchini18 I believe this is caused by the same bug I filed earlier this week: #5838. Additional output counts do not account for count simulation, hence the discrepancy.

**Reply by luchini18** [reply]
Ah yes you're right this looks similar, thanks.
